Engine/Commands/InvokeScriptAnalyzerCommand.cs

```diff
@@ -34,7 +34,9 @@ public class InvokeScriptAnalyzerCommand : PSCmdlet, IOutputWriter
 
         #region Private variables
         List<string> processedPaths;
-        private int totalDiagnosticCount = 0;
+        // initialize to zero for all severity enum values
+        private Dictionary<DiagnosticSeverity, int> diagnosticCounts =
+                Enum.GetValues<DiagnosticSeverity>().ToDictionary(s => s, _ => 0);
```

Comment:
Ugh it looks like this doesn't work in net462:

```
error CS0308: The non-generic method 'Enum.GetValues(Type)' cannot be used with type arguments 
```

so annoying...but based on my read of https://learn.microsoft.com/en-us/dotnet/api/system.enum.getvalues?view=net-9.0 I think you could do this with:

```
Enum.GetValues(DiagnosticSeverity).ToDictionary(s => (DiagnosticSeverity)s, _ => 0);
```

and that should work in all versions. Instead of the lovely builtin that just gets the strings for us, gotta get the enum values (which is just gonna be ints 0-3 I think) and then cast them back to the name...


scratch that I think you can just use `Enum.GetNames(DiagnosticSeverity)` per https://learn.microsoft.com/en-us/dotnet/api/system.enum.getnames?view=netframework-4.6.2

Reply:
@MatejKafka did you want to try this out?

Reply:
Thanks, should be fixed now.

Reply:
Did `Enum.GetNames(DiagnosticSeverity)` not work?

Reply:
`.GetNames` returns strings, we want the values here, since they're easier to work with.